paddle/framework/recurrent_network_op.h

```diff
@@ -0,0 +1,200 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#pragma once
+
+#include <google/protobuf/text_format.h>
+#include "paddle/framework/attr_checker.h"
+#include "paddle/framework/ddim.h"
+#include "paddle/framework/enforce.h"
+#include "paddle/framework/scope.h"
+#include "paddle/framework/variable.h"
+
+// Remove when including operator.h
+#include "paddle/framework/attr_checker.h"
+#include "paddle/framework/op_desc.pb.h"
+
+namespace paddle {
+namespace framework {
+
+// --------------------------------------------------------------------
+// fake interfaces that has not be implemented by other modules.
+// TODO keep updating according to other modules' designs.
+struct OpRunContext {
+  Scope* scope;
+};
+
+// TODO replace this with Net's proto.
+struct NetDesc {
+  std::string name_;
+};
+
+class PlainNet {
+ public:
+  PlainNet() {}
+  PlainNet(const NetDesc& desc) {}
+  PlainNet(const std::string desc) {}
+  void Run(Scope* scope) {}
+};
+
+class OperatorBase {
+ public:
+  virtual ~OperatorBase() {}
+  void Init(const OpDesc& op_desc, AttributeMap& attrs) {}
+  virtual void Run(OpRunContext* context) const = 0;
+  virtual void InferShape(const Scope* scope) const = 0;
+  inline Variable* Input(Scope* scope, int index) const {
+    return scope->GetVariable(inputs_[index]);
+  };
+
+  template <typename T>
+  inline const T GetAttr(const std::string& name) const {
+    return boost::get<T>(attrs_.at(name));
+  }
+
+ protected:
+  std::vector<std::string> inputs_;
+  std::vector<std::string> outputs_;
+  AttributeMap attrs_;
+};
+// fake interfaces end
+// --------------------------------------------------------------------
+// TODO:
```

Comment:
The sequence format in RecurrentOp is Tensor<seq_len, batch_size, dim> now, we consider all the sequences have the same length. After this simplest version, we will design new sequence format like sequenceStartPositions of original Paddle.